api/src/main/java/com/inrupt/client/ProblemDetails.java

```diff
@@ -0,0 +1,146 @@
+
+    public ProblemDetails(
+        final URI type,
+        final String title,
+        final String details,
+        final int status,
+        final URI instance
+    ) {
+        this.type = type;
+        this.title = title;
+        this.details = details;
+        this.status = status;
+        this.instance = instance;
+    }
+
+    public URI getType() {
+        return this.type;
+    };
+
+    public String getTitle() {
+        return this.title;
+    };
+
+    public String getDetails() {
+        return this.details;
+    };
+
+    public int getStatus() {
+        return this.status;
+    };
+
+    public URI getInstance() {
+        return this.instance;
+    };
+
+    public static class ProblemDetailsData {
+        public URI type;
+        public String title;
+        public String details;
+        public int status;
+        public URI instance;
+    }
+
+    private static JsonService getJsonService() {
+        if (ProblemDetails.isJsonServiceInitialized) {
+            return ProblemDetails.jsonService;
+        }
+        // It is a legitimate use case that this is loaded in a context where no implementation of the JSON service is
+        // available. On SPI lookup failure, the ProblemDetails exceptions will fall back to default and not be parsed.
+        try {
+            ProblemDetails.jsonService = ServiceProvider.getJsonService();
+        } catch (IllegalStateException e) {
+            ProblemDetails.jsonService = null;
+        }
+        ProblemDetails.isJsonServiceInitialized = true;
+        return ProblemDetails.jsonService;
+    }
+
+    public static ProblemDetails fromErrorResponse(
+            final int statusCode,
+            final Headers headers,
+            final byte[] body
+    ) {
+        final JsonService jsonService = getJsonService();
+        if (jsonService == null
+                || (headers != null && !headers.allValues("Content-Type").contains(ProblemDetails.MIME_TYPE))) {
+            return new ProblemDetails(
+                URI.create(ProblemDetails.DEFAULT_TYPE),
+                null,
+                null,
+                statusCode,
+                null
+            );
+        }
+        try {
+            final ProblemDetailsData pdData = jsonService.fromJson(
+                    new ByteArrayInputStream(body),
+                    ProblemDetailsData.class
+            );
+            final URI type = Optional.ofNullable(pdData.type)
+                .orElse(URI.create(ProblemDetails.DEFAULT_TYPE));
+            // JSON mappers map invalid integers to 0, which is an invalid value in our case anyway.
+            final int status = Optional.of(pdData.status).filter(s -> s != 0).orElse(statusCode);
+            return new ProblemDetails(
+                    type,
+                    pdData.title,
+                    pdData.details,
+                    status,
+                    pdData.instance
+            );
+        } catch (IOException e) {
```

Comment:
Unfortunately no, more specific exceptions are Jackson or JSON-B specific. Our abstraction layer doesn't define a specific exception beyond `IOException` (it might be a future improvement though, in the spirit of the `ClientHttpException`)